pkg/cmd/check/check_test.go

```diff
@@ -0,0 +1,87 @@
+package check
```

Comment:
Always use check_test for testing, encourages not testing internal implementation.

Reply:
How about we do the following:
* Separate internal tests to a separate file, as they are useful for debugging and understanding what a function does (white box)
* have a separate file in a separated package for end user testing (black box)

Then both benefits can be reaped?

Reply:
Wont you be able to do that by changing ur input to the exported functions (after all the exported functions call the unexported functions), right?

Reply:
In this example, I think it can be done without testing unexported functions - the  `NewCmdCheckSemantic` can be used to test by calling the execute function in cobra.